doc/auto_evo.md

```diff
@@ -0,0 +1,75 @@
+# Auto-Evo
+
+Auto-Evo is not a single algorithm, but a longstanding problem at the center of
+thrive. The goal of Auto-Evo is for Thrive to be able to generate new creatures
+to populate the world through the evolutionary process in real time. This
+problem comes with a few constraints. As already mentioned, Auto-Evo must be
+able to run in real time on all of our player's PCs. Additionally, Auto-Evo must
+generate creatures that have sufficient variety to provide an interesting
+experience for the player, while still sticking to expectations of a viable
+creature.
+
+## The Miche Algorithm
+While Auto-Evo is not in itself an algorithm, it is solved by one. The current
+algorithm used to solve Auto-Evo is the Miche Algorithm.
+
+### Miche
+The term Miche comes from a shortening of Multi-Niche (and a nod to computer
+science professor Dr. Meesh.) Each Miche is a node in a tree structure called
+the Miche Tree. Each of these Miches contains a list of child Miches,
+recursively creating the tree. If a node does not contain any child nodes it is
+known as a Leaf Miche, and instead contains a Species known as the Occupant.
+Every Miche additionally contains a Selection Pressure further explained below.
+
+### Selection Pressure
+A Selection Pressure is a function that provides a fitness score to a species.
+Selection Pressures measure fitness in a specific attribute, eg, being able to
+osmoregulate or the ability to find food. Fitness Scores are arbitrary and
+specific to each pressure (With the exception that a score <= 0 means a pressure
+has been failed.) Therefore scores are always compared relative to another
+species via WeightedComparedScores. Each selection pressure contains a list of
+Mutation Strategies that are applied to try and increase the Pressure's score in
+the mutations.
+
+### Mutation Strategies
+A Mutation Strategy is a function that creates new mutated species from a parent
+species. Strategies are often very specific, eg add an organelle that produces
+ATP from iron or remove an organelle that consumes glucose. These allow Auto-Evo
+to only attempt to generate and try a specific subset of the possible mutations
+that are each much more likely to be an improvement on the parent. Each Mutation
+Strategy also has an MP cost to limit the amount of mutations per generation.
+
+### Steps
+Each time Auto-Evo is run it goes through a series of steps.
+
+### Generate Miche Step
```

Comment:
I think these are sub-sections of "steps" so these should have `####` to be a smaller heading.